### PR TITLE
trojan scanner

### PR DESCRIPTION
**Pull request by frenchy64**
Related https://github.com/advthreat/iroh/issues/6052

Scans repository every build, and uberjar before every deployment.

There's something wrong with CTIA's uberjar. Worked around it and logged here: https://github.com/advthreat/iroh/issues/6060

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: trojan scanner
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: scripts/uberjar-trojan-scan.clj [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+          (println "var in `./scripts/uberjar-trojan-scan.clj`.")
+          1))))
+
+(defn unzip-uberjar []
+  (let [_ (check (process ["rm" "-rf" scanning-directory] {:inherit true}))
+        _ (check (process ["mkdir" "-p" scanning-directory] {:inherit true}))
+        ;; FIXME logged: https://github.com/advthreat/iroh/issues/6060
+        ;; erm, there's something wrong with ctia's META-INF/license directory,
+        ;; it cannot be unzipped with jar.
+        ;;   java.io.IOException: META-INF/license : could not create directory
```
Comment (agzam) [inline scripts/uberjar-trojan-scan.clj:84]: Looks like a permission issue 🤷

File: scripts/actions/install-trojansourcedetector.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+set -Eeuxo pipefail
+
+if ! command -v trojansourcedetector &> /dev/null 
+then
+  rm -rf tmp/trojansourcedetector
+  mkdir -p tmp/trojansourcedetector
+  cd tmp/trojansourcedetector
+  curl -sLO https://github.com/haveyoudebuggedit/trojansourcedetector/releases/download/v1.0.1/trojansourcedetector_1.0.1_linux_amd64.tar.gz
```
Comment (agzam) [inline scripts/actions/install-trojansourcedetector.sh:10]: What happens if it fails to fetch it?

**Review by agzam — APPROVED** [review]
Very cool how you were able to get it done so quickly. Lack of docstrings/comments a bit inconvenient, but it feels that if someone starts poking around it, it wouldn't be too difficult to understand what's happening.

File: scripts/actions/install-trojansourcedetector.sh (hunk 2 above)
Comment (frenchy64) [reply scripts/actions/install-trojansourcedetector.sh:10]: `set -e` fails the script.

File: scripts/do_not_edit-uberjar_trojansourcedetector.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    "goog/i18n/datetimepatternsext.js",
+    "goog/i18n/datetimesymbols.js",
+    "goog/i18n/datetimesymbolsext.js",
+    "goog/labs/i18n/listsymbols.js",
+    "goog/labs/i18n/listsymbolsext.js",
+    "goog/i18n/datetimepatterns.js",
+    "goog/format/internationalizedemailaddress.js",
+    "swagger-ui/swagger-ui-bundle.js"
+  ]
+}
```
[the remark this answers is not in the document]
Comment (frenchy64) [reply scripts/do_not_edit-uberjar_trojansourcedetector.json:33]: Start with an empty vector, iterate and add exclusions for things that seemed plausible to need bidi chars.

File: scripts/actions/install-trojansourcedetector.sh (hunk 2 above)
Comment (agzam) [reply scripts/actions/install-trojansourcedetector.sh:10]: Ah of course! 👍

File: scripts/uberjar-trojan-scan.clj (hunk 1 above)
Comment (frenchy64) [reply scripts/uberjar-trojan-scan.clj:84]: Here's how to reproduce:

```
lein uberjar
cd target
jar xf ctia.jar
```

**Pull request merged**